consensus/scp/src/node.rs

```diff
@@ -138,34 +132,29 @@ pub trait ScpNode<V: Value>: Send {
     /// Get local node quorum set.
     fn quorum_set(&self) -> QuorumSet;
 
-    /// Submit a list of values of nomination.
-    fn nominate(
-        &mut self,
-        slot_index: SlotIndex,
-        values: BTreeSet<V>,
-    ) -> Result<Option<Msg<V>>, String>;
+    /// Submit a list of values for nomination.
+    fn nominate(&mut self, values: BTreeSet<V>) -> Result<Option<Msg<V>>, String>;
 
     /// Handle incoming message from the network.
     fn handle(&mut self, msg: &Msg<V>) -> Result<Option<Msg<V>>, String>;
 
     /// Get externalized values (or an empty vector) for a given slot index.
-    fn get_externalized_values(&self, slot_index: SlotIndex) -> Vec<V>;
-
-    /// Check if we have externalized values for a given slot index.
-    fn has_externalized_values(&self, slot_index: SlotIndex) -> bool;
+    fn get_externalized_values(&self, slot_index: SlotIndex) -> Option<Vec<V>>;
 
     /// Process pending timeouts.
     fn process_timeouts(&mut self) -> Vec<Msg<V>>;
 
-    /// Get metrics for a specific slot.
-    fn get_slot_metrics(&mut self, slot_index: SlotIndex) -> Option<SlotMetrics>;
+    /// Get the current slot's index.
+    fn current_slot_index(&self) -> SlotIndex;
+
+    /// Get metrics for the current slot.
```

Comment:
This seems like it might be problematic since node auto-advances... could we add the slot_index and keep a vector of metrics around, similar to externalized slots?

Reply:
I am still concerned about this